components/autofill/src/sync/engine.rs

```diff
@@ -100,29 +102,33 @@ impl<T: SyncRecord + std::fmt::Debug> SyncEngine for ConfigSyncEngine<T> {
         self.put_meta(&tx, LAST_SYNC_META_KEY, &(timestamp.as_millis() as i64))?;
 
         incoming_impl.finish_incoming(&tx)?;
-        // Not quite sure if we should commit now and then stage outgoing?
-        tx.commit()?;
 
         // Finally, stage outgoing items.
-        // TODO: Call yet-to-be-implemented stage outgoing code
-        //     let outgoing = self.fetch_outgoing_records(timestamp)?;
-        //     Ok(outgoing)
-        Ok(OutgoingChangeset::new(self.collection_name(), timestamp))
+        let outgoing = outgoing_impl.fetch_outgoing_records(
+            &tx,
+            self.config.collection.to_string(),
+            timestamp,
+        )?;
+        // Not quite sure if we should commit now and then stage outgoing?
```

Comment:
We should - so maybe change this comment to something like "it may take a long time to actually perform the upload and we've already staged everything we need to complete the sync in a way that doesn't need it to stay alive, so commit now and start a new one once complete"